From f4aae9c101895a30e3e3033c822fec2ea27922ce Mon Sep 17 00:00:00 2001
From: Remi Gacogne <remi.gacogne@powerdns.com>
Date: Mon, 29 Aug 2016 15:50:44 +0200
Subject: [PATCH] auth: Fix a stack-based off-by-one write in the HTTP remote
 backend

Reported by mongo (thanks!).

(cherry picked from commit 9e5fa399eea5152a451753f1db68dec46537447c)
---
 modules/remotebackend/httpconnector.cc | 1 -
 1 file changed, 1 deletion(-)

Index: pdns/modules/remotebackend/httpconnector.cc
===================================================================
--- pdns.orig/modules/remotebackend/httpconnector.cc
+++ pdns/modules/remotebackend/httpconnector.cc
@@ -376,7 +376,6 @@ int HTTPConnector::recv_message(Json& ou
           throw NetworkError("EOF while reading");
         if (rd<0)
           throw NetworkError(std::string(strerror(rd)));
-        buffer[rd] = 0;
         arl.feed(std::string(buffer, rd));
       }
       // timeout occured.
